code/asteroid/asteroid.cpp

```diff
@@ -159,42 +206,27 @@ static float asteroid_cap_speed(int asteroid_info_index, float speed)
 }
 
 /**
- * Returns whether position is inside inner bounding volume
- *
- * Sum together the following: 1 inside x, 2 inside y, 4 inside z
- * inside only when sum = 7
+ * Check if asteroid is within inner bound
+ * @return 0 if not inside or no inner bound, 1 if inside inner bound
  */
-static int asteroid_in_inner_bound_with_axes(asteroid_field *asfieldp, vec3d *pos, float delta)
-{
-	Assert(asfieldp->has_inner_bound);
+static bool asteroid_in_inner_bound(asteroid_field *asfieldp, vec3d *pos, float delta) {
 
-	int rval = 0;
-	if ( (pos->xyz.x > asfieldp->inner_min_bound.xyz.x - delta) && (pos->xyz.x < asfieldp->inner_max_bound.xyz.x + delta) ) {
-		rval += 1;
-	}
-
-	if ( (pos->xyz.y > asfieldp->inner_min_bound.xyz.y - delta) && (pos->xyz.y < asfieldp->inner_max_bound.xyz.y + delta) ) {
-		rval += 2;
-	}
-
-	if ( (pos->xyz.z > asfieldp->inner_min_bound.xyz.z - delta) && (pos->xyz.z < asfieldp->inner_max_bound.xyz.z + delta) ) {
-		rval += 4;
-	}
+	if (!asfieldp->has_inner_bound)
+		return false;
 
-	return rval;
+	return (pos->xyz.x > asfieldp->inner_min_bound.xyz.x - delta) && (pos->xyz.x < asfieldp->inner_max_bound.xyz.x + delta) &&
+		(pos->xyz.y > asfieldp->inner_min_bound.xyz.y - delta) && (pos->xyz.y < asfieldp->inner_max_bound.xyz.y + delta) &&
+		(pos->xyz.z > asfieldp->inner_min_bound.xyz.z - delta) && (pos->xyz.z < asfieldp->inner_max_bound.xyz.z + delta);
 }
 
-/**
- * Check if asteroid is within inner bound
- * @return 0 if not inside or no inner bound, 1 if inside inner bound
- */
-static int asteroid_in_inner_bound(asteroid_field *asfieldp, vec3d *pos, float delta) {
+static bool asteroid_ship_is_inside_field(asteroid_field* asfieldp, vec3d* pos, float radius) {
 
-	if (!asfieldp->has_inner_bound) {
-		return 0;
-	}
+	radius *= 2.0f;
 
-	return (asteroid_in_inner_bound_with_axes(asfieldp, pos, delta) == 7);
+	return pos->xyz.x + radius > Asteroid_field.min_bound.xyz.x && pos->xyz.x - radius < Asteroid_field.max_bound.xyz.x&&
+		   pos->xyz.y + radius > Asteroid_field.min_bound.xyz.y && pos->xyz.y - radius < Asteroid_field.max_bound.xyz.y&&
```

Comment:
So far the only thing that I'm pretty sure should be changed. At the end of these lines there really should be a space after and x and y for readability.